### PR TITLE
Support 'Clerk Training' mode

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

- Added 'CLERK TRAINING' store to Airtable
- Added 'Summer Strawberry' as customer in AirTable
- Prefills 'Summer Strawberry''s phone number on `CustomerLookupScreen` if clerk signed in is `Sunny Citrus`
- Transactions don't get logged / points don't get updated in `CheckoutScreen` if current store is 'CLERK TRAINING' (handled in `confirmTransaction`)

Not really logically part of the same PR, but also updated:
- Transaction "ID" (last seven digits/numbers) shown on Confirmation Screen (^:
- Confirmation Screen: styling/spacing updated

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Online sources

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

#12 
#19 

## How to review

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

~Need to update the customer side to filter out the 'Clerks Training' store from the store list!~ addressed in https://github.com/calblueprint/dccentralkitchen/pull/44

Also not logically part of the PR, but realized edge case bugs existed: 
- `ClerkLoginScreen` and `CustomerLookupScreen` where state is not reset upon nav. back to a mounted screen
- Attempted to address using by adding `componentDidUpdate` and `_reset` functions to both, then realized this was incorrect implementation https://github.com/react-navigation/react-navigation/issues/4867
- Solution will be to upgrade to `react-navigation` version 5 https://reactnavigation.org/blog/2020/02/06/react-navigation-5.0/ and handle in https://reactnavigation.org/docs/use-focus-effect/

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots

image

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: screens/CheckoutScreen.js [hunk 1]
```diff
@@ -168,6 +168,18 @@ export default class CheckoutScreen extends React.Component {
     return msg;
   };
 
+  createFakeTransaction = transactionInfo => {
```
Comment (anniero98) [inline screens/CheckoutScreen.js]: Can we move this to `checkoutUtils`

File: screens/StoreLookupScreen.js [hunk 2]
```diff
@@ -83,6 +83,9 @@ export default class StoreLookupScreen extends React.Component {
   };
 
   handleNavigate = () => {
+    if (this.state.store.storeName === 'CLERK TRAINING') {
+      console.log('settrainingmodetotruebutidkhow');
```
Comment (anniero98) [inline screens/StoreLookupScreen.js]: You'll need to keep this in AsyncStorage !!

File: screens/CheckoutScreen.js [hunk 3]
```diff
@@ -168,6 +168,18 @@ export default class CheckoutScreen extends React.Component {
...
+    const fakeTransaction = {
```
Comment (anniero98) [inline screens/CheckoutScreen.js]: You can just write this as `{...transactionInfo, id: 'TRAINED', totalPrice: transactionInfo.totalSale}`

Nit: I don't know if 'TRAINED' seems like the best thing to display on Confirmation?

File: lib/checkoutUtils.js [hunk 4]
```diff
@@ -82,3 +82,14 @@ export function displayDollarValue(amount, positive = true) {
...
   return '-'.concat(dollarValue);
 }
+export function createFakeTransaction(transactionInfo) {
+  return {
+    ...transactionInfo,
+    id: 'ABCXYZ',
+    pointsEarned: transactionInfo.pointsEarned,
+    rewardsApplied: transactionInfo.rewardsApplied,
+    subtotal: transactionInfo.subtotal,
+    discount: transactionInfo.discount,
```
Comment (tommypoa) [inline lib/checkoutUtils.js]: Unnecessary with line 87.

File: screens/CheckoutScreen.js [hunk 5]
```diff
@@ -170,6 +176,14 @@ export default class CheckoutScreen extends React.Component {
 
   // Adds the transaction to the user's account and updates their points.
   confirmTransaction = async transactionInfo => {
+    // Clerk Training: if storeId is "Clerk Training"'s ID
+    // do not create the transaction or update points
+    // Navigate to Confirmation Screen with a pre-filed transaction ID
+    const storeId = await AsyncStorage.getItem('storeId');
+    if (JSON.parse(await AsyncStorage.getItem('trainingMode'))) {
+      this.props.navigation.navigate('Confirmation', createFakeTransaction(transactionInfo));
```
Comment (tommypoa) [inline screens/CheckoutScreen.js]: Code nit: name your prop and use it correspondingly in `ConfirmationScreen` for consistency with transactionId prop and clarity.

File: screens/CustomerLookupScreen.js [hunk 6]
```diff
@@ -17,17 +17,28 @@ export default class CustomerLookupScreen extends React.Component {
...
   async componentDidMount() {
-    const clerkName = await AsyncStorage.getItem('clerkName');
-    this.setState({ clerkName, phoneNumber: '', errorMsg: null });
+    await this._reset();
+    this.setState({ loading: false });
   }
 
+  _reset = async () => {
+    const clerkName = await AsyncStorage.getItem('clerkName');
+    // Clerk Training: pre-fill customer (Summer Strawberry) phone number
```
Comment (tommypoa) [inline screens/CustomerLookupScreen.js]: The pre-filling only happens the first time the component is mounted, and is no longer pre-filling when the clerk finishes checkout and proceeds to next customer. Not sure if this is the intended behaviour - flagging in case it isn't!

File: screens/ConfirmationScreen.js [hunk 7]
```diff
@@ -35,6 +35,11 @@ export default class ConfirmationScreen extends React.Component {
     this.props.navigation.navigate('CustomerLookup');
   };
 
+  writeToClipboard = copyText => {
```
Comment (anniero98) [inline screens/ConfirmationScreen.js]: very fun, nice!

File: styled/checkout.js [hunk 8]
```diff
@@ -8,7 +8,7 @@ import { ColumnContainer, RowContainer } from './shared';
 export const TopBar = styled.View`
   height: 68px;
   width: 100%;
-  background: ${Colors.base};
+  background: ${props => (props.trainingColor ? '#FFC300' : Colors.base)};
```
Comment (anniero98) [inline styled/checkout.js]: For consistency, can we add this to `Colors.js`? `Colors.training` (LOL)?

Comment (anniero98) [reply styled/checkout.js]: update: I am making this `Colors.trainingMode`

**Comment by anniero98** [comment]
@wangannie this is ready, so please do some sanity-checking; if everything seems ok we can merge it
